### ISSUE TITLE
Native Query cannot be cancelled

### ISSUE THREAD

**Issue by Usiel** [metabase/metabase#11713]
**Describe the bug**
Native Query cannot be cancelled

**To Reproduce**
1. Create a long running native query
2. Run the query
3. Click on abort
4. Nothing happens

**Expected behavior**
Query should be cancelled

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-GB",
    "platform": "Linux x86_64",
    "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/79.0.3945.117 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "11.0.5+10",
    "java.vendor": "AdoptOpenJDK",
    "java.vendor.url": "https://adoptopenjdk.net/",
    "java.version": "11.0.5",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "11.0.5+10",
    "os.name": "Linux",
    "os.version": "4.14.47-56.37.amzn1.x86_64",
    "user.language": "en",
    "user.timezone": "GMT"
  },
  "metabase-info": {
    "databases": [
      "postgres"
    ],
    "hosting-env": "elastic-beanstalk",
    "application-database": "postgres",
    "run-mode": "prod",
    "version": {
      "date": "2020-01-13",
      "tag": "v0.34.1",
      "branch": "release-0.34.x",
      "hash": "265695c"
    },
    "settings": {
      "report-timezone": "Asia/Kuala_Lumpur"
    }
  }
}
```

**Severity**
Annoying if one is tweaking a long running query and has to wait until the previous execution finishes.

**Additional context**
Adding PR in a second. Not quite sure if I'm missing anything to actually cancel the query on database side, but it seems to work similar to non-native queries.

**Comment by aejnsn** [comment]
It's probably a bit late to offer any further elaboration on this given the above PR. Let's say you run a long-running query, and then change it in such a way that fixes the slow nature of it, then run again with Cmd+Enter. You get the results for your more immediate query, however once the original slow query completes, the slow results replace the results of the immediate query. Should this perhaps be a separate issue or does this fix also encompass that situation as well?

**Comment by paulrosenzweig** [comment]
@aejnsn I just opened a [separate issue](https://github.com/metabase/metabase/issues/11727) for that. Thanks for bringing it up!

**Comment by paulrosenzweig** [comment]
Fixed by #11714
